chore: Update ClerkProvider page (#374)
clerkinc/clerk-docs: docs/components/clerk-provider.mdx

---
title: <ClerkProvider />
description: The <ClerkProvider /> component wraps your React application to provide active session and user context to Clerk's hooks and other components.
---

import { Tab, Tabs } from "nextra-theme-docs";

> import { CodeBlockTabs } from "@/components/CodeBlockTabs";

Reviewer: Remove imports - unnecessary

> ## Props

Reviewer: Are all of these optional? Or just some?
Because I believe they are all optional, and in that case, we can remove the `?` from each prop, and add copy that says:

```suggestion
## Props

All props below are optional.
```

Author: Some of them are required, most notably the `publishableKey` or `frontendApi` value

> | Name                      | Type                                       | Description                                                                                                                                                                                                                |
> |---------------------------|--------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

Reviewer: There's a lot of extra spacing in this table.. I'd probably refactor it to have normal spacing like the other tables.

```suggestion
| Name | Type | Description |
| --- | --- | --- |
```

Author: The way the table is formatted right now is how Prettier formats tables (see [this playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAJOBDAJnATgHSgF4TTDCB9AGQggGsAaCgAkh2YCMAbCMOgZ2YZccZgDMIuALYYY8LMwBUMWgEJF5KAAMdzAFb9CYgK5QwMAJbRmFnLAswAngAoAHgEpmwZiJjHcUMyuzAC+hDpamgAKFjwwAD4Agha4AA6SCagQ-oZQALR58UgFSPElhABSEAAWgQAiEHDxAMoA0gDi8QCMAMw9AByVGFCiAEqN8RUAYq3xPV0ALJp5zCtrqxvrW5ubmgDUzFQW-DCEzAcA7g7VQswA2lwWUHQAuszOAPQqH-wQUnAXap4ODuQgHYYKO7DCAwIG4ZjQOAvTRne7Q2F4BEjF5IZjMapyVJID4fOCuDBSVJcOAAOkgUmYAHIAKLkynU5iWGDUxmaGgiBkWVL8YwMrAQHjw-gOIT-GAMVjQfhwcxwPzw7BC45gJ4Ac2YcEeMBphAAwv4MBwHP5FVBlar1cwZK4LFJjIJcMd3QrlQpna73ZzDVxA5YzLZTDADcaQAwQBBUpYlchQMJcBALlFhAh+MgQBguBcMI5c3GOLgMHw1c0KXAjiNkGIC8qyxWqzBmqlK3rkDBcMY4HHlVILL3+4OQNKoLrqQBFYwwuCN5sTgyuZp6ucL+DLrgtkAAR23cCi6eFeYw-DyIzgOCwsZAfYwsT1pr+MgvXC4D6nM7giTkT0OGMeAojweslyQJs9wnAkpC4AB1aoHDgfguzAOBmhzBwLAANwcRw8zAfhSxAXCBwASSgOwOzAT1E0Sajmicald33VJ02VBCK1SPMONQvBcKXOMnmVXAYFPDBdQ-KCVzjLtcDEvMZFwOhxQuKAHw4p4YAQ2xYWQfoAAY4xEI8UhPCtpIwNiJxgS09KwAykAAJjjd04AAFUtXNZJguM4CkDhbzvKhhl1YwpLgKZJBkOQexQDAQIgEAQhCIA)). So once we introduce linting/formatting into our workflow the tables will be formatted like that normally.

You can configure it tho, see https://github.com/prettier/prettier-vscode/issues/762 for context

Reviewer: Oh amazing!

| `clerkJSVariant?`         | `string`                                   | If your web application only uses Control components you can set this value to `headless` and load a minimal ClerkJS bundle for optimal page performance.                                                                  |
| `clerkJSVersion?`         | `string`                                   | Define the npm version for `@clerk/clerk-js`.                                                                                                                                                                              |
| `supportEmail?`           | `string`                                   | Optional support email for display in authentication screens. Will only affect Clerk Components and not Account Portal pages.                                                                                              |
| `localization?`           | `object`                                   | Optional object to localize your Components. Will only affect Clerk Components and not Account Portal pages.                                                                                                               |

> | `appearance?`             | `object`                                   | Optional object to customize your Components. Will only affect Clerk Components and not Account Portal pages.                                                                                                              |

Reviewer: ```suggestion
| `appearance` | <code>[Appearance](/docs/components/customization/overview) \| undefined</code> | Optional object to style your components. Will only affect [Clerk Components][components-ref] and not [Account Portal][ap-ref] pages. |

```

| `allowedRedirectOrigins?` | `Array<string \| RegExp>`                  | Optional array of domains used to validate against the query param of an auth redirect. If no match is made, the redirect is considered unsafe and the default redirect will be used with a warning passed to the console. |
| `afterSignInUrl`          | `string`                                   | Full URL or path to navigate after successful sign in.                                                                                                                                                                     |

> | `afterSignUpUrl`          | `string`                                   | Full URL or path to navigate after successful sign up.                                                                                                                                                                     |

Reviewer 2: @LekoArts Should we add `isSatellite`, `proxyUrl`, and `domain` ?

Author: Yes, we should. But our internal docs are sparse on this and call it experimental so I left it out. But feel free to put up a PR or nudge the person who would know more to also add it in another PR :)

Reviewer 3: We reference `isSatellite` and `domain` on https://clerk.com/docs/advanced-usage/satellite-domains so we can grab the info from there and add it here for consistency if nothing else.

Reviewer 2: @LekoArts I've updated the codebase and removed the experimental tags.

> | `afterSignUpUrl`          | `string`                                   | Full URL or path to navigate after successful sign up.                                                                                                                                                                     |

Reviewer: ```suggestion
| `afterSignUpUrl`          | `string`                                   | Full URL or path to navigate after successful sign up.                                                                                                                                                                     |

[components-ref]: /docs/components/overview
[ap-ref]: /docs/account-portal/overview
```
